lib/tasks/integration.rake

```diff
@@ -92,51 +12,205 @@ task :integrate, :log do |_, args|
 
     line
   end
+end
 
-  require 'ci_reporter_shell'
 
-  report = CiReporterShell.report('tmp/junit')
 
-  junit = Dir['tmp/junit/**']
+def resolve_test_directories
+
+  directories_requiring_to_run_in_different_job_or_without_tests = %w{test_helpers fixtures shoulda_macros factories remote performance}.map {|x| "test/#{x}"}
+
+  test_dirs = Pathname.new("test").children.select(&:directory?).map(&:to_s) - directories_requiring_to_run_in_different_job_or_without_tests
+
+end
+
+def test_commands
+    tags_for_test_categories = %w{
+      @backend
+      @emails
+      @stats
+      @search
+      @no-txn
+    }
+
+    test_dirs = resolve_test_directories
+
+    return {
+      :cucumber_javascript => 'parallel_cucumber --verbose features -o "-b -p parallel --tags=@javascript --tags=~@fakeweb --tags=~@percy"',
+      :cucumber_non_tagged => %Q{parallel_cucumber --verbose features -o "-b -p parallel --tags=~@javascript #{tags_for_test_categories.map {|t| %Q|--tags=~#{t}|}.join(' ')}"},
+      :cucumber_for_categories => %Q{parallel_cucumber --verbose features -o "-b -p parallel --tags=~@javascript --tags=#{tags_for_test_categories.join(',')}"},
+
+      :rspec => 'parallel_rspec --verbose spec',
+      :integration => "parallel_test --verbose #{test_dirs.delete('test/integration')}",
+      :swagger => [
+        'rake doc:swagger:validate:all',
+        'rake doc:swagger:generate:all',
+      ],
+      :frontend => [
+        'rake ci:jspm --trace',
+        'yarn test -- --reporters dots,junit --browsers Firefox',
+        'yarn jest',
+        'rake db:purge db:setup',
+      ],
+      :functional => "parallel_test --verbose #{test_dirs.delete('test/functional')}",
+      :license_checks => "export http_proxy=#{ENV['http_proxy']} https_proxy=#{ENV['https_proxy']}; rake ci:license_finder:run",
+      :main_suite => "parallel_test --verbose #{test_dirs.join(' ')}",
+      :percy => 'PERCY_ENABLE=1 cucumber -b -p parallel --tags=@percy features',
+    }
+
+end
 
-  if junit.present?
-    puts 'WARNING: tmp/junit is not empty'
-    puts junit
-    abort 'Will not continue, tmp/junit is not empty'
-  end
 
-  total_time = 0
 
-  results = tasks.map do |task|
+
+
+desc 'The default execution: the whole CI suite'
+task :integrate => 'integrate:parallel'
+
+
+namespace :integrate do
```

Comment:
Probably under `:test` namespace though `:integrate` is good too

Reply:
i'll stay on :integrate for now, to limit scope of changes